ee/tabby-db/src/email_service_credential.rs

```diff
@@ -0,0 +1,91 @@
+use anyhow::Result;
+use rusqlite::OptionalExtension;
+
+use crate::DbConn;
+
+const MAIL_CREDENTIAL_ROW_ID: i32 = 1;
+
+#[derive(Debug, PartialEq)]
+pub struct EmailServiceCredentialDAO {
+    pub smtp_username: String,
+    pub smtp_password: String,
+    pub smtp_server: String,
+}
+
+impl EmailServiceCredentialDAO {
+    fn new(smtp_username: String, smtp_password: String, smtp_server: String) -> Self {
+        Self {
+            smtp_username,
+            smtp_password,
+            smtp_server,
+        }
+    }
+}
+
+impl DbConn {
+    pub async fn get_email_service_credential(&self) -> Result<Option<EmailServiceCredentialDAO>> {
+        let res = self
+            .conn
+            .call(|c| {
+                Ok(c.query_row(
+                    "SELECT smtp_username, smtp_password, smtp_server FROM email_service_credential WHERE id=?",
+                    [MAIL_CREDENTIAL_ROW_ID],
+                    |row| Ok(EmailServiceCredentialDAO::new(row.get(0)?, row.get(1)?, row.get(2)?)),
+                )
+                .optional())
+            })
+            .await?;
+        // Unsure why the map_err is needed. The `?` from the previous line
+        // should convert it automatically, but this breaks without it.
+        res.map_err(Into::into)
```

Comment:
could you try diff this with GithubOAuthCredentialDAO (maybe line by line) to debug?